openedx/features/qverse_features/registration/signals.py

```diff
@@ -1,20 +1,182 @@
+    if created:
+        LOGGER.info('{} user has been created.'.format(edx_user.username))
+    else:
+        edx_user.first_name = user_info.get('firstname')
+        edx_user.last_name = user_info.get('surname')
+        edx_user.email = user_info.get('email')
+        edx_user.save()
+        LOGGER.info('{} user has been updated.'.format(edx_user.username))
+    return edx_user
+
+
+def _create_or_update_edx_user_profile(edx_user):
+    """
+    Creates/Updates edx user profile.
+
+    Arguments:
+        edx_user (User): An edx user instance whose profile is going to be created/updated
+    """
+    full_name = '{} {}'.format(edx_user.first_name, edx_user.last_name)
+    _, created = UserProfile.objects.update_or_create(user=edx_user, defaults={'name': full_name})
+    if created:
+        LOGGER.info('{} edx profile has been created.'.format(edx_user.username))
+    else:
+        LOGGER.info('{} edx profile has been updated.'.format(edx_user.username))
+
+
+def _create_or_update_qverse_user_profile(edx_user, user_info, output_file_rows):
+    """
+    Creates/Updates qverse user profile.
+
+    It also stores the status of the creation/update in the receiving list which
+    contains the content which will be written on output file.
+
+    Arguments:
+        edx_user (User): An edx user instance whose qverse profile is going to be created/updated
+        user_info (dict): User info whose profile is going to be created/updated
+        output_file_rows (list): List of CSV file rows which will be written on output file
+    """
+    department = Department.objects.get(number=user_info.get('departmentid'))
+    qverse_profile_data = {
+                            'current_level': user_info.get('levelid'),
+                            'other_name': user_info.get('othername'),
+                            'mobile_number': user_info.get('mobile'),
+                            'programme': user_info.get('programmeid'),
+                            'department': department,
+                            'registration_number': user_info.get('regno')
+                            }
+    _, created = QVerseUserProfile.objects.update_or_create(user=edx_user,
+                                                            defaults=qverse_profile_data)
+    if created:
+        LOGGER.info('{} qverse profile has been created.'.format(edx_user.username))
+        output_file_rows[-1].update({'status': 'Created'})
+    else:
+        LOGGER.info('{} qverse profile has been updated.'.format(edx_user.username))
+        output_file_rows[-1].update({'status': 'Updated'})
+
+
+def _write_status_on_csv_file(filename, output_file_rows):
+    """
+    Writes on the given file.
+
+    Arguments:
+        filename (str): Complete path of the output file
+        output_file_rows (list): List of CSV file rows
+    """
+    try:
+        with open(filename, 'w') as csv_file:
+            if len(output_file_rows) > 0:
```

Comment:
this can be simply written as `if output_file_rows:`

Reply:
done.